HHS/Head-Start-TTADP
# [Prod] Allow admins to unblock training report regional point of contacts; fix bug with "some goals are incomplete" message

## Description of change
- Fixes a bug where certain goals would not display as incomplete in the "some goals are incomplete" message on the activity reports
- Give admins the ability to undo the POC "I've completed my work" checkbox
- Also adds a tool for snapshotting and restoring a database at the beginning and end of writing a test. Meant to reduce test "flakiness" going forward.

## How to test
- Create FEI goals for two recipients. Complete the root causes for one, leave the other blank. Add both recipients to an AR, add the FEI goal to the AR, and confirm that the review and submit screen shows "some goals are incomplete" and links to the correct goal for correction
- Mark complete the POC part of the session as a POC user. Then, as an admin, unlock that section and confirm it can now be updated again by the POC user.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3053
* https://jira.acf.gov/browse/TTAHUB-2787
* https://jira.acf.gov/browse/TTAHUB-3035

## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
